OmeroPy py3 update

will-moore opened the pull request:
In progress...

will-moore committed:
Start work on developers/Python.rst page

will-moore committed:
print(text) fixes on Python.rst

will-moore committed:
long and Bytes fixes to Python.rst

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-docs#91](https://idr1-slot3.openmicroscopy.org:48443/jenkins/job/OMERO-docs/91/). See the [console output](https://idr1-slot3.openmicroscopy.org:48443/jenkins/job/OMERO-docs/91/consoleText) for more details.
Possible conflicts:
  - PR #2010 will-moore 'Add annotation examples from saeri scripts'
    - omero/developers/Python.rst
  - PR #2011 will-moore 'Python example script links'
    - omero/developers/Python.rst

--conflicts

joshmoore commented:
@will-moore : what do you want to do about #2010?

will-moore commented:
I'd still like to get #2010 merged, so I'll list it for final review

will-moore committed:
Python.rst shows pip install steps

I removed the instructions for running the examples because I can't imagine
that very many users go through this potentially tricky process

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-docs#200](https://latest-ci.openmicroscopy.org/jenkins/job/OMERO-docs/200/). See the [console output](https://latest-ci.openmicroscopy.org/jenkins/job/OMERO-docs/200/consoleText) for more details.
Possible conflicts:
  - PR #2010 will-moore 'Add annotation examples from saeri scripts'
    - omero/developers/Python.rst
  - PR #2011 will-moore 'Python example script links'
    - omero/developers/Python.rst

--conflicts

will-moore committed:
Merge remote-tracking branch 'origin/develop' into omeropy_py3_update

joshmoore committed:
Remove conflict marker

joshmoore commented on omero/developers/Python.rst:
```diff
@@ -1,66 +1,33 @@
+From OMERO 5.6.0 release, the client library ``omero-py`` supports Python 3 and
+can be installed via ``pip``. We recommend you use a virtual environment:
+
+    $ python3 -m venv py3_venv
```
These sections need to be prefaced with:

```
::

    $ python3 ...
```

will-moore committed:
fix code formatting

will-moore commented on omero/developers/Python.rst:
```diff
@@ -1,66 +1,35 @@
+
+    $ python3 -m venv py3_venv
+    $ source py3_venv/bin/activate
+    $ pip install omero-py
```
As discussed: need to use the omero-py dev version variable here.

will-moore committed:
pip install |version_py| on Python.rst

joshmoore commented on omero/conf_autogen.py:
```diff
@@ -13,6 +13,6 @@
 version_dsl_plugin = "5.5.0"
 version_blitz_plugin = "5.5.0"
 version_ice_builder = "1.5.0"
-version_py = "5.5.dev2"
```
This is handled via the autogen step generally, but bumping manually doesn't hurt. :+1:
